File: src/test/live_sub.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

/**
 * This tutorial demonstrates simple receipt of messages over the ROS system.
 */

void chatter_cb(const std_msgs::String::ConstPtr& msg)
{
  ROS_INFO("I heard: [%s]", msg->data.c_str());
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");

  ros::NodeHandle n;
  std::string topic;
  n.getParam( "/chatter_topic", topic );
  ros::Subscriber sub = n.subscribe(topic, 1000, chatter_cb);

  ros::spin();

  return 0;
}

File: src/test/live_sub.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def callback(data):
    rospy.loginfo(rospy.get_caller_id()+"I heard %s",data.data)
    
def listener():
    rospy.init_node('listener', anonymous=True)
    try:
      topic = rospy.get_param( "~chatter_topic" )
      rospy.loginfo( "Listening to [%s]" % topic )
      rospy.Subscriber(topic, String, callback)
      rospy.spin()
    except:
      print( "Failed to get parameters!" )
        
if __name__ == '__main__':
    listener()
